### DISCUSSION TITLE
Google cloud nightscout VM suddenly charging costs for "Compute Engine"

### DISCUSSION THREAD

**Discussion by emp-00** [NightscoutFoundation/xDrip#3061] [Q&A] [answered]
Dear team, I'm running a NS instance on Google Cloud as per @Navid200 's instructions already since 11 months (30 more days till 1 year). Everything was running smoothly and no cost incurred at all **until 3 days ago**. End of August some strange errors were logged in the VM logs - and the costs are clearly related to "Compute Engine" usage (not network traffic). I have only NS running on the VM and have changed nothing in the configuration in the last 6 months at least.

For fire-fighting, yesterday I logged into the NS instance, updated the platform/packages via the xDrip-team-menu and rebooted the VM. However, it looks like costs are still registered even though I cannot see any real high CPU/memory etc. load in my trends. See below some screenshots... Can anybody make sense out of this situation? What's happening? Anybody else having or had the same issue? Thanks so much!

PS: We're talking € 0.03 in 2 days (!) - I'm not concerned about the 3 cents but I worry about this trend!

![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/f7cccb81-354d-462e-b155-9c9159b2e389)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/0f7c16a8-f28c-4fd0-b77d-fd7f84ff0f5f)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/2f967105-526c-4f61-9995-5e96f873572f)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/d14eed28-1d01-4846-9239-85a54d2d0610)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/63bf194a-d46f-4ec5-9035-127e20f3e39d)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/a26ac3d1-5ffd-4082-be31-b94ad683e583)

**Comment by Navid200** [comment, marked as answer]
This comes up a lot.  That's why I have this page:
https://navid200.github.io/xDrip/docs/Nightscout/NS_FreeTier.html

It tells you to create a cost table.
https://navid200.github.io/xDrip/docs/Nightscout/CostTable.html  
I will improve the cost table page in the guide today.

When you open the cost table, for every major cost at the top, you will see an identical discount (negative) at the bottom.
Ignoring all of those, you may be left with a few positive values at the top.  Those are the cause of you having been charged.
Please create a cost table and show what those are.

**Reply by emp-00** [reply]
Thanks @Navid200 - this new help-page is very helpful to lighten up the Google cloud "cost djungle". However, the "cost table" only allows to select the PAST month / PAST billing period (here: August), but not the current month (September). So any currently accumulating costs cannot be tracked with this method.

Anyway - in my case in August it was (for me, for the first time!) indeed Chinese network traffic, just as you're explaining in your guide. This was not occurring for 11 months but now my site seems to have made it into Chinese crawler lists :-( ... I agree, 1 cent may not be worth the time to program a blocker, but still, it would be great to block this traffic via a firewall rule.

For September I'm not sure if I will also only see 1 cent, because I'm already at 3 cent within 2 days of September, see above.

![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/47cc61f3-2c7a-456d-9001-ffce5bb37864)

![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/a49914d7-63e7-4c5b-8e30-614f163b9c43)

**Reply by Navid200** [reply]
That's correct, the cost table can only be used for the past months.  It is meant to clarify what the cost is for instead of relying on speculation.  

It's not the time required to add a block.  It is how much more complicated our code will become, which we will need to keep maintaining  and test every time we make a change.  That's why we have not done it not because we don't want to spend the time creating the block.

You may be misunderstanding the cost table.  Your compute engine cost for last month was $3.67, not $0.01.
I am improving the cost table page in the guide hoping to encourage people to take advantage of it for their understanding.

**Reply by emp-00** [reply]
Yeah, last month is clear (thanks to your guide) - it was Chinese traffic, all understood. However, this month is September and as you can see, my Billing Overview page already displays 3 cents for only 2 days of September. Is this a misinterpretation?

**Reply by Navid200** [reply]
Yes, that cost is for compute engine, which is a part of free tier.  Can you see any free tier credit in your current month?
The credits only show up at the end of the month.  That's why it is pointless to worry looking at the current month cost.

**Reply by emp-00** [reply]
Well, it the current month's September "[BILLING ACCOUNT OVERVIEW]" balance. It would be very strange (for me) if this "one click cost overview" balance turns out to be completely wrong. What do you see here as of today in your account?

**Reply by Navid200** [reply]
A negative cost is a credit.  
  
![265222198-a49914d7-63e7-4c5b-8e30-614f163b9c43](https://github.com/NightscoutFoundation/xDrip/assets/51497406/05bfe623-9a8d-4ce0-9e3d-bdeb04b0d1be)
